lib/Parser/Parse.cpp

```diff
@@ -2447,6 +2523,146 @@ IdentPtr Parser::ParseImportOrExportFromClause(bool throwIfNotFound)
     return moduleSpecifier;
 }
 
+template<bool buildAST> 
+ParseNodePtr Parser::ParseDefaultExportClause()
+{
+    Assert(m_token.tk == tkDEFAULT);
+
+    m_pscan->Scan();
+    ParseNodePtr pnode = nullptr;
+    ushort flags = fFncNoFlgs;
+    
+    switch (m_token.tk)
+    {
+    case tkCLASS:
+        {
+            if (!m_scriptContext->GetConfig()->IsES6ClassAndExtendsEnabled())
+            {
+                goto LDefault;
```

Comment:
@abchatra here is a reason why I'd prefer we remove this feature flags once we enable a feature.  Otherwise we need to make sure the feature flags are maintained and kept functionally correct when newer features come online.  @boingoing is doing that here, but in general either we need to make sure we are disciplined about ensuring the switches remain functionally correct (or at least don't crash), or we abandon them.